src/test/java/wolox/training/controller/BookControllerTest.java

```diff
@@ -0,0 +1,104 @@
+package wolox.training.controller;
+
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.utils.JsonUtil;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class BookControllerTest {
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Autowired
+	private BookRepository bookRepository;
+
+	private Book bookBuilder;
+
+	@BeforeEach
+	public void init() {
+		bookBuilder = new Book();
+		bookBuilder.setGenre("M");
+		bookBuilder.setImage("xxxx");
+		bookBuilder.setTitle("avengers");
+		bookBuilder.setPublisher("NA");
+		bookBuilder.setYear("1990");
+		bookBuilder.setPages(1);
+		bookBuilder.setSubtitle("NA");
+	}
+
+	@Test
+	public void whenCreateBook_thenFindAllUsers() throws Exception {
+		bookBuilder.setAuthor("author1");
+		bookBuilder.setIsbn("isbn1");
+		bookRepository.save(bookBuilder);
+
+		mockMvc.perform(get("/api/books").contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk())
+				.andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+				.andExpect(jsonPath("$[0].author", is("author1")));
+	}
+
+	@Test
+	public void whenCreateBook_thenFindByAuthor() throws Exception {
+
+		bookBuilder.setAuthor("author2");
+		bookBuilder.setIsbn("isbn2");
+		bookRepository.save(bookBuilder);
```

Comment:
Here you are using another component that you can mock the functionality thus not having a dependency with them (ie, by using Mockito and declaring `private BookRepository bookRepository` with the `@MockBean` annotation).